fdbserver/workloads/ConsistencyCheck.actor.cpp

```diff
@@ -344,48 +351,85 @@ struct ConsistencyCheckWorkload : TestWorkload {
 					} else {
 						rangeToCheck = allKeys;
 					}
-				}
+					TraceEvent("ConsistencyCheck_Config")
+					    .setMaxEventLength(-1)
+					    .setMaxFieldLength(-1)
+					    .detail("RangeToCheck", rangeToCheck)
+					    .detail("Indefinite", self->indefinite)
+					    .detail("Distributed", self->distributed)
+					    .detail("ClientCount", self->clientCount)
+					    .detail("ClientId", self->clientId)
+					    .detail("PerformQuiescentChecks", self->performQuiescentChecks)
+					    .detail("PerformTSSCheck", self->performTSSCheck)
+					    .detail("PerformCacheCheck", self->performCacheCheck);
+					state std::vector<std::pair<KeyRange, Value>> shardLocationPairListForDistributed =
+					    wait(self->getKeyLocationsForRangeList(cx, { allKeys }, self));
+					// Check that each failed shard has the same data on all storage servers that it resides on
+					wait(::success(self->checkDataConsistency(cx,
+					                                          shardLocationPairListForDistributed,
+					                                          configuration,
+					                                          tssMapping,
+					                                          self,
+					                                          /*consistencyCheckEpoch=*/0)));
 
-				TraceEvent("ConsistencyCheck_Config")
-				    .setMaxEventLength(-1)
-				    .setMaxFieldLength(-1)
-				    .detail("RangeToCheck", rangeToCheck)
-				    .detail("Distributed", self->distributed)
-				    .detail("ClientCount", self->clientCount)
-				    .detail("ClientId", self->clientId);
-
-				bool keyServerResult =
-				    wait(self->getKeyServers(cx, self, keyServerPromise, rangeToCheck.withPrefix(keyServersPrefix)));
-				if (keyServerResult) {
-					state std::vector<std::pair<KeyRange, std::vector<StorageServerInterface>>> keyServers =
-					    keyServerPromise.getFuture().get();
-
-					// Get the locations of all the shards in the database
-					state Promise<Standalone<VectorRef<KeyValueRef>>> keyLocationPromise;
-					bool keyLocationResult = wait(self->getKeyLocations(cx, keyServers, self, keyLocationPromise));
-					if (keyLocationResult) {
-						state Standalone<VectorRef<KeyValueRef>> keyLocations = keyLocationPromise.getFuture().get();
-
-						// Check that each shard has the same data on all storage servers that it resides on
-						wait(::success(self->checkDataConsistency(cx, keyLocations, configuration, tssMapping, self)));
-
-						// Cache consistency check
-						if (self->performCacheCheck)
-							wait(::success(self->checkCacheConsistency(cx, keyLocations, self)));
+				} else {
+					// Get a list of key servers; verify that the TLogs and master all agree about who the key servers
```

Comment:
Just a note: this comment is probably wrong, which seems to be there before.

Reply:
Agree. I will go through the single version later.